### PR TITLE
Refactor rg-header to work with ui-router and state. Removes complexi…

### PR DESCRIPTION
**Pull request by gregbenner**
Refactor rg-header to work with ui-router and state. Removes complexity of transcluding attributes in favour of simple angular models. Update demo and tests.

### CODE REVIEW

**Comment by nwhite89** [comment]
@gregbenner no merge on this, you need to be able to set `is-active="true"` not just using `sref` this is a feature for Create and can't be removed can you please ensure that this is still possible

File: src/rg-header/rg-header-item.tpl.html [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <li class="RgHeader-navItem">
-    <a class="RgHeader-navItemInner" ng-transclude ng-class="{'is-active': isActive === 'true'}">
```
Comment (nwhite89) [inline src/rg-header/rg-header-item.tpl.html]: is active is a must needed.

Comment (gregbenner) [inline src/rg-header/rg-header-item.tpl.html]: For sure can be added back, won't conflict. Didn't see it used in the demo wasn't sure.

Comment (nwhite89) [inline src/rg-header/rg-header-item.tpl.html]: Can you add to the demo a version using active and one using `ui-sref-active` please cheers

Comment (gregbenner) [inline src/rg-header/rg-header-item.tpl.html]: Not sure how it functions as it never worked for us on Team Distribute.
I can add with ui-sref however it requires an added state and will show up as a component in the list on localhost:7000

Comment (nwhite89) [inline src/rg-header/rg-header-item.tpl.html]: not sure how what worked the `is-active`? you had an attribute you set as true 
```
<rb-header-item is-active="true"></rb-header-item>
```

**Comment by nwhite89** [comment]
yeah go for it cheers :+1:

**Comment by gregbenner** [comment]
Ready from my point of view for merge but please also check this if you want to merge https://github.com/rockabox/rbx_campaign_management/pull/268 should be done in parallel.

**Comment by gregbenner** [comment]
@nwhite89 I ended up resolving with ng-if to avoid conflicts between href and lack of ui-sref, should be good to go.

**Pull request merged**